### PR TITLE
net-misc/sx: new ebuild

### PR DESCRIPTION
**Pull request by tomboy-64**
@aballier this doesn't depend on any dev-ml stuff, so it can go already.

### CODE REVIEW

File: net-misc/sx/sx-1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit eutils
+
+DESCRIPTION="Skylable SX - a distributed object-storage software for data clusters."
```
Comment (mgorny) [inline net-misc/sx/sx-1.2.ebuild]: This is not a full sentence so no dot at the end.

Comment (tomboy-64) [inline net-misc/sx/sx-1.2.ebuild]: If that's your only complaint, I'm more than happy to oblige :)

File: net-misc/sx/sx-1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		ewarn "Have a nice day."
+		ewarn "********************************************************************************"
+	fi
+}
+
+src_configure() {
+	econf --prefix=/usr/ \
+          --sysconfdir=/etc/ \
+          --enable-dependency-tracking \
+          --disable-silent-rules \
```
Comment (mgorny) [inline net-misc/sx/sx-1.2.ebuild]: Any reason you repeat all the defaults? ;-P

**Comment by mgorny** [comment]
The bot's complaints are a bigger problem here ;-P.

File: net-misc/sx/sx-1.2.ebuild (hunk 2 above)
Comment (tomboy-64) [inline net-misc/sx/sx-1.2.ebuild]: yup. want to make it obvious for later editors. i really hate it if i dig up an ebuild to edit later and have to dig through all the default docs.

**Comment by tomboy-64** [comment]
i expect removing ~ppc64 would solve this?
why didn't `repoman all` catch this?

**Comment by tomboy-64** [comment]
@gentoo-repo-qa-bot test this please

**Comment by mgorny** [comment]
It's got no ears, it can't hear you. Just give him ~20 minutes to notice the new push.

File: net-misc/sx/sx-1.2.ebuild (hunk 2 above)
Comment (mgorny) [inline net-misc/sx/sx-1.2.ebuild]: That's not a good reason at all. All ebuilds default to those prefixes, so repeating them is really wrong.

Comment (aballier) [inline net-misc/sx/sx-1.2.ebuild]: and also kills EPREFIX support :)

**Comment by tomboy-64** [comment]
@aballier @mgorny mostly rewrote it. please review.

**Comment by aballier** [comment]
where did my comments go???

**Comment by aballier** [comment]
fails tests here:
```
make[4]: Entering directory '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server'
FAIL: test/run-nginx-test.sh
PASS: test/hdist-test
============================================================================
Testsuite summary for sx 1.2
============================================================================
# TOTAL: 2
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
```

`/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/test/run-nginx-test.sh.log` contains:

```
make[6]: Entering directory '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sxscripts'
 /bin/mkdir -p '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/etc'
 /bin/mkdir -p '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/etc/sxserver'
 /usr/lib/portage/python2.7/ebuild-helpers/xattr/install -c -m 644  sxserver/sxhttpd.conf.default sxserver/fastcgi_params '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/etc/sxserver'
 /bin/mkdir -p '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/etc/sxserver/errors'
 /usr/lib/portage/python2.7/ebuild-helpers/xattr/install -c -m 644  sxserver/errors/414.json sxserver/errors/50x.json '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/etc/sxserver/errors'
 /bin/mkdir -p '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/sbin'
 /usr/lib/portage/python2.7/ebuild-helpers/xattr/install -c sbin/sxserver sbin/sxsetup sbin/sxdump '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sx-test-wzQ6qS0i/sbin'
 /bin/mkdir -p '/usr/share/doc/sx-1.2'
/bin/mkdir: cannot create directory ‘/usr/share/doc/sx-1.2’: Permission denied
Makefile:391: recipe for target 'install-nobase_docDATA' failed
make[6]: *** [install-nobase_docDATA] Error 1
make[6]: Leaving directory '/var/tmp/portage/net-misc/sx-1.2/work/sx-1.2/server/sxscripts'
```

**Comment by tomboy-64** [comment]
Weird. I ran it this afternoon and it worked.

**Comment by aballier** [comment]
you might want to try running tests after unmerging sx I think

**Comment by tomboy-64** [comment]
Did that. The path wasn't there when I ran the test. I also tried creating it and making `chmod a-rwx /usr/share/doc`. No chance of reproducing it.

**Comment by aballier** [comment]
got it, use:
```
--docdir="\${prefix}/share/doc/${PF}" 
```

and add a comment before econf saying that tests make a temporary install relative to $prefix, so doc dir must be relative to it also

**Comment by tomboy-64** [comment]
Any idea why that asploded with you and not with me?

**Comment by aballier** [comment]
no clue why this was ok with you, maybe FEATURES=userpriv that I use?

${prefix} is where autoconf stores its --prefix option; so, since econf uses EPREFIX properly, it should be ok

and here doc was dumped in /usr/share/doc/sx-1.2/ ; got to try again

**Comment by tomboy-64** [comment]
oh, it's simple. i'm being stupid.

When i tested with `ebuild sx-1.2.ebuild clean ... install` everything went fine. With `emerge -1 -j1 --nodeps sx` I found no problems ... simply because I still had an old version of sx in my local overlay.

Sorry for the work; gonna push the changes in a few minutes :(

**Comment by tomboy-64** [comment]
don't merge this please, something went wrong here with other commits.

**Comment by tomboy-64** [comment]
okay, good to go.

**Pull request merged**